src/app/modules/form/admin-form/admin-form.controller.ts

```diff
@@ -2775,7 +2774,7 @@ export const handleSetGoLinkSuffix: ControllerHandler<
             },
           ),
           // TODO: fix error handling (https://linear.app/ogp/issue/FRM-901/improve-error-handling-when-calling-gogov-api)
-          () => new ApplicationError('Error occurred when claiming GoGov link'),
+          () => new GoGovError(),
```

Comment:
Since this PR is closing FRM-901, shall we break down the errors further here? https://guide.go.gov.sg/developer-guide/api-documentation#errors. 
200 -> 200
400 -> 400
429 -> 429
401/402/404/5xx -> 500

Reply:
Nope, this PR won't be closing [FRM-901](https://linear.app/ogp/issue/FRM-901), will do that in another PR. Also, should we be lumping 401/402/404/5xx -> 500?

Reply:
I think yes for 401, and 404. Since those errors are something that we (FormSG -> GoGov) have a contract with. But for the 400, and 429, it's something that's a contract between User and FormSG.

`401`: is an invalid API key, which is something that's not fixable by the user. It is a config issue on FormSG BE.
`404`: is also a contract between FormSG and GoGov where it is FormSG that determines the resource.

`402`: Might be interesting. What do you guys think?

Reply:
imo `402` could map to `502`, since FormSG is making API calls to Go on behalf of the user and acting as a proxy

Reply:
Makes sense

Reply:
will do this in a separate PR!